src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -909,6 +909,15 @@ internal int CalculateSyntaxOffsetInSynthesizedConstructor(int position, SyntaxT
                 return syntaxOffset;
             }
 
+            if (declaration.Declarations.Length == 1 && position == declaration.Declarations[0].Location.SourceSpan.Start)
```

Comment:
Is it necessary that `Declarations.Length == 1`? What if the type has multiple partial declarations?